vm/src/value.rs

```diff
@@ -439,7 +439,23 @@ impl<'a, 't> InternalPrinter<'a, 't> {
             Value::Userdata(ref data) => arena.text(format!("{:?}", data)),
             Value::Thread(thread) => arena.text(format!("{:?}", thread)),
             Value::Byte(b) => arena.text(format!("{}", b)),
-            Value::Int(i) => arena.text(format!("{}", i)),
+            Value::Int(i) => {
+                use base::types::BuiltinType;
+                match **self.typ {
+                    Type::Builtin(BuiltinType::Int) => arena.text(format!("{}", i)),
+                    Type::Builtin(BuiltinType::Char) =>
+                        if 0 <= i && i <= ::std::u32::MAX as isize {
+                            match ::std::char::from_u32(i as u32) {
+                                Some('"') => arena.text(format!("'{}'", '"')),
+                                Some(c) => arena.text(format!("'{}'", c.escape_default())),
+                                None => unreachable!(),
```

Comment:
Could you change this to  something akin to `ìce!("Invalid character passed to pretty printing")` (`ice!` reports in internal compiler error with a link to gluon's issue list).

(Using `unreachable!` is fine but I think it is best to limit it to places where it is possible to locally verify that the branch is `unreachable`. Here the character can be created far away from this branch)

Reply:
Is it really an internal compiler error though? This is more of a runtime problem, no?

Reply:
Well the "compiler" (speaking about it in a broader sense) has made a mistake causing an invalid character to be stored. Is there any reason to separate "Internal compiler errors" from "Internal virtual machine errors"?